### PR TITLE
put namespace objects first in the output

### LINKED ISSUE #65: Respect ordering in kustomization resources array
It can be extremely useful for namespace creation. Namespace have to be created before all objects you want to place inside. Example:

```shell
$ cat kustomization.yaml 
commonLabels:
  app: hello
resources:
- namespace.yaml
- cm.yaml

$ kustomize build .

apiVersion: v1
data:
  hello.txt: |
    "hello"
kind: ConfigMap
metadata:
  labels:
    app: hello
  name: hello-config-ck54t29m9k
  namespace: hello
---
apiVersion: v1
kind: Namespace
metadata:
  labels:
    app: hello
  name: hello

$ kustomize build . | kubectl apply -f -      
namespace "hello" created
Error from server (NotFound): error when creating "STDIN": namespaces "hello" not found
```

Namespace manifest is situated at the end of stdout, which cause that error message.

### PR DESCRIPTION
**Pull request by Liujingfang1**
Fix #65

### CODE REVIEW

File: pkg/resmap/idslice_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		{i: 3, j: 0},
+		{i: 3, j: 1},
+		{i: 2, j: 3},
+	}
+
+	for _, ind := range indexes {
+		if !ids.Less(ind.i, ind.j) {
+			t.Fatalf("%+v should be in front of %+v", ids[ind.i].Gvk(), ids[ind.j].Gvk())
+		}
+	}
```
Comment (droot) [inline pkg/resmap/idslice_test.go]: Another way to test would be invoking `sort.Sort(ids)` and compare the output with the expected ordering using reflect pkg.

Comment (Liujingfang1) [reply pkg/resmap/idslice_test.go]: I'm going to use `sort`. It seems to have better readability. Thank you :)

Comment (Liujingfang1) [reply pkg/resmap/idslice_test.go]: This is update. PTAL

File: pkg/resmap/idslice_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		resource.NewResId(schema.GroupVersionKind{Kind: "Namespace"}, "ns1"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "Namespace"}, "ns2"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "ConfigMap"}, "cm"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "Pod"}, "pod"),
+	}
+	sort.Sort(ids)
+	if !reflect.DeepEqual(ids, expected) {
+		t.Fatalf("expected %+v but got %+v", expected, ids)
+	}
+}
```
Comment (droot) [inline pkg/resmap/idslice_test.go]: Awesome!!!

**Comment by tkellen** [comment]
Thank you!

**Pull request merged**